### PR TITLE
[stable/kong] add load balancer source range parameters

### PR DESCRIPTION
**Pull request by thirstydeveloper**
#### What this PR does / why we need it:

Adds helm value parameters for specifying source CIDR ranges to restrict access to the proxy / admin service when the service type is LoadBalancer.

Of particular interest for restricting access to the admin interface as part of a defense in depth approach.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/kong/ci/loadbalancer-values.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# CI test for LoadBalancer admin/proxy types
+
```
Comment (hbagdi) [inline stable/kong/ci/loadbalancer-values.yaml:2]: Can we introduce a `default-values.yaml` to run the chart with the default values?

Comment (thirstydeveloper) [reply stable/kong/ci/loadbalancer-values.yaml:2]: Oops - I thought the test script would still run a version with `values.yaml`.

I just added a symlink from `ci/default-values.yaml` -> `values.yaml`, in the hopes that will work so there won't be two copies of the same file to maintain.

If that doesn't work... I guess duplication is the answer.

File: stable/kong/README.md [hunk 2]
```diff
@@ -72,6 +73,7 @@ and their default values.
 | proxy.nodePort                    | Node port when service type is `NodePort`                              |                       |
 | proxy.type                        | k8s service type. Options: NodePort, ClusterIP, LoadBalancer           | `NodePort`            |
 | proxy.loadBalancerIP              | To reuse an existing ingress static IP for the admin service           |                       |
+| proxy.loadBalancerSourceRanges    | Limit admin access to CIDRs if set and service type is `LoadBalancer`  | `[]`                  |
```
Comment (hbagdi) [inline stable/kong/README.md]: The description needs update here, it seems like a copy paste error.

**Review by hbagdi — COMMENTED** [review]
@thirstydeveloper One minor comment. Could you resolve the conflict? I'll approve and merge then.
Thank you for contributing this!

**Comment by thirstydeveloper** [comment]
/hold

I think the syntax isn't quite right in the service template files. I need some time to resolve.

File: stable/kong/README.md (hunk 2 above)
Comment (thirstydeveloper) [reply stable/kong/README.md]: Done!

**Comment by thirstydeveloper** [comment]
/hold cancel

Should be good now.

File: stable/kong/templates/service-kong-admin.yaml [hunk 3]
```diff
@@ -13,9 +13,17 @@ metadata:
     heritage: "{{ .Release.Service }}"
 spec:
   type: {{ .Values.admin.type }}
-  {{- if and (eq .Values.admin.type "LoadBalancer") .Values.admin.loadBalancerIP }}
+  {{- if eq .Values.admin.type "LoadBalancer" }}
+  {{- if .Values.admin.loadBalancerIP }}
   loadBalancerIP: {{ .Values.admin.loadBalancerIP }}
   {{- end }}
+  {{- if .Values.admin.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:
```
Comment (thirstydeveloper) [inline stable/kong/templates/service-kong-admin.yaml:21]: I'm new to the go templating language so if there's a better way to format this I'm totally open to suggestions.

Comment (hbagdi) [reply stable/kong/templates/service-kong-admin.yaml:21]: This looks good. The last one indeed gave an error as it was expanding as an array directly. Thanks for fixing this!

**Comment by hbagdi** [comment]
@thirstydeveloper Thank you for contributing this feature!

**Pull request merged**